libsolidity/analysis/TypeChecker.cpp

```diff
@@ -113,18 +113,24 @@ void TypeChecker::checkDoubleStorageAssignment(Assignment const& _assignment)
 	size_t storageByteArrayPushes = 0;
 	size_t storageByteAccesses = 0;
 	auto count = [&](TupleExpression const& _lhs, TupleType const& _rhs, auto _recurse) -> void {
-		TupleType const& lhsType = dynamic_cast<TupleType const&>(*type(_lhs));
-		TupleExpression const* lhsResolved = dynamic_cast<TupleExpression const*>(resolveOuterUnaryTuples(&_lhs));
+		auto const& lhsType = dynamic_cast<TupleType const&>(*type(_lhs));
+		auto const* lhsResolved = dynamic_cast<TupleExpression const*>(resolveOuterUnaryTuples(&_lhs));
+
+		if (!lhsResolved) {
+			m_errorReporter.reportErrors("lhsResolved is null");
+			return;
+		}
```

Comment:
Another think that is missing here is adding an entry about the bugfix to the changelog.

Like I said, please take a look at the [review checklist](https://github.com/ethereum/solidity/blob/develop/ReviewChecklist.md). It lets you find out these small issues on your own so that we don't have to point them out each time in review :)